consensus/safety-rules/src/persistent_storage.rs

```diff
@@ -97,30 +115,50 @@ fn test_in_memory_storage() {
 
 pub struct OnDiskStorage {
     file_path: PathBuf,
+    file_path_alt: PathBuf,
     internal_data: InMemoryStorage,
 }
 
 impl OnDiskStorage {
     pub fn new_storage(file_path: PathBuf) -> Result<Box<dyn PersistentStorage>> {
-        let internal_data = InMemoryStorage::load_config(file_path.clone())?;
-        Ok(Box::new(Self {
-            file_path,
-            internal_data,
-        }))
+        Self::new(file_path, false)
     }
 
     pub fn default_storage(file_path: PathBuf) -> Result<Box<dyn PersistentStorage>> {
-        if file_path.exists() {
-            return Self::new_storage(file_path);
+        Self::new(file_path, true)
+    }
+
+    fn new(mut file_path: PathBuf, default: bool) -> Result<Box<dyn PersistentStorage>> {
```

Comment:
It seems that this commit is independent from the rest of the changes in this PR. Could we move it into a separate PR? (unless I'm mistaken you can have it on top of master)